### PR TITLE
DVDDemuxFFmpeg: Discard repair track from GoPro recordings

### PR DESCRIPTION
**Pull request by popcornmix**
GoPro video files always stutter at the start (e.g. for 20 seconds) and then resume fine.
https://forum.kodi.tv/showthread.php?tid=259292
https://trac.kodi.tv/ticket/17371
https://trac.kodi.tv/ticket/17373

MrMC had a fix and in testing that resolves the reported issue.
https://forum.kodi.tv/showthread.php?tid=259292&pid=2622828#pid2622828

## Description
Video (mp4) from GoPro cameras can have a 'meta' track used for a file repair containing
'fdsc' data, this is also called the SOS track. Ignore it or we will glitch bad on startup.

## Types of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp [hunk 1]
```diff
@@ -1443,6 +1443,10 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
       {
         stream = new CDemuxStream();
         stream->type = STREAM_DATA;
+        // Video (mp4) from GoPro cameras can had a 'meta' track used for a file repair containing
+        //  'fdsc' data, this is also called the SOS track. Ignore it or we will glitch bad on startup.
+        if (pStream->codec->codec_tag == MKTAG('f','d','s','c'))
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: adding a discareded stream is not really what you want to achieve, right? better not adding it to the map at all

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: @FernetMenta do you mean this?
```
        stream->type = STREAM_NONE;
```

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: nope, AddStream adds a stream to a map of available streams for the demuxer that is queried by player. two lines above a new CDemuxStream object is created that you try to invalidate. I mean don't create it for this case.

Comment (popcornmix) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: Can you explain exactly what is desired here?
Returning nullptr from this function causes playback to fail 
```
17:03:30.360 T:1700598560   ERROR: CDVDDemuxFFmpeg::AddStream - internal error, stream is null
```

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: do not allocate a packet and return empty (bReturnEmpty) in this case

**Comment by popcornmix** [comment]
Superceded by https://github.com/xbmc/xbmc/pull/13095

**Pull request closed without merging**